freshmaker/handlers/botas/botas_shipped_advisory.py

```diff
@@ -65,12 +75,90 @@ def handle(self, event):
+                log.warning("Bundle %s doesn't have 'bundle_path_digests' set",
+                            bundle['bundle_path'])
+                continue
+            bundle_digests.add(bundle['bundle_path_digest'])
+        bundle_images = self._pyxis.get_images_by_digests(bundle_digests)
+
+        # now we have nvrs of bundles to rebuild
+        bundles_nvrs = self._filter_bundles_by_pinned_related_images(
+            bundle_images, original_images_digests)
 
         # Skip that event because we can't proceed with processing it.
         # TODO
-        # Next step would be queries Pyxis and getting the digests of these nvrs
+        # Now when we have bundle images' nvrs we can procceed with rebuilding it
+        msg = f"Skipping the rebuild of {len(bundles_nvrs)} bundle images " \
+              "due to being blocked on further implementation for now."
         db_event.transition(EventState.SKIPPED, msg)
         db.session.commit()
         return []
+
+    def _filter_bundles_by_pinned_related_images(self, bundle_images, digests):
+        """
+        If the digests were not pinned by OSBS, the bundle image
+        will be filtered out.
+
+        There is no need in checking 'created_by_osbs' value, because we already
+        know that digest points to the manifest list, because of previous
+        filtering.
+
+        :param list bundle_images: ContainerImages of operator bundles
+        :param set digests:  Digests of the original images that were rebuilt
+            that should be in 'related_images' and be 'pinned'
+        :return: list of NVRs of bundle images that have at least one
+            original related image that was rebuilt
+        """
+        ret_bundle_images_nvrs = set()
+        with koji_service(conf.koji_profile, log, dry_run=conf.dry_run,
+                          login=False) as session:
+            for bundle in bundle_images:
+                nvr = bundle['brew']['build']
+                build = session.get_build(nvr)
+                related_images = (
+                    build.get("build", {})
+                         .get("extra", {})
+                         .get("image", {})
+                         .get("operator_manifests", {})
+                         .get("related_images", {})
+                )
+
+                for pullspec in related_images.get('pullspecs', []):
+                    related_image_digest = self.parse_digest(pullspec['new'])
+                    if related_image_digest is None or\
+                            related_image_digest not in digests:
+                        continue
+
+                    # this is one of the original images,
+                    # check if OSBS-pinning is used for this image
+                    if pullspec.get("pinned", False):
+                        # add bundle nvr to a set for later handling
+                        ret_bundle_images_nvrs.add(nvr)
+                        break
```

Comment:
@apaplaus I think you misunderstood me. This is what I meant in [r521574574](https://github.com/redhat-exd-rebuilds/freshmaker/pull/38/files#r521574574).
```suggestion
                # Skip the bundle if the related images section was not populated by OSBS
                if related_images.get("created_by_osbs") is not True:
                    continue

                ret_bundle_images_nvrs.add(nvr)
```

Reply:
Ohh I thought you meant it in opposite way, that I suppose to get rid of 'creadted_by_osbs' check. And actually I don't understand well what we are doing here but I will add your suggested change.

Reply:
@apaplaus this is because you've already determined through Pyxis that the bundle image has an operator/operand image that matches the digest of the manifest list of the original operator/operand image. We therefore just need to check if OSBS created the related images section in the bundle image. Each pullspec has the `pinned` value and that gets set if the pullspec referenced in the bundle image was not already a digest. If the bundle image maintainer used a digest instead of a floating tag, OSBS will set `pinned` to `False`.